pkg/telemetrylogs/condition_builder.go

```diff
@@ -218,15 +233,15 @@ func (c *conditionBuilder) ConditionFor(
 	operator qbtypes.FilterOperator,
 	value any,
 	sb *sqlbuilder.SelectBuilder,
-    _ uint64,
-    _ uint64,
+	_ uint64,
+	_ uint64,
 ) (string, error) {
 	condition, err := c.conditionFor(ctx, key, operator, value, sb)
 	if err != nil {
 		return "", err
 	}
 
-	if operator.AddDefaultExistsFilter() {
+	if !(strings.HasPrefix(key.Name, BodyJSONStringSearchPrefix) && constants.BodyV2QueryEnabled) && operator.AddDefaultExistsFilter() {
```

Comment:
nit: maybe add a differnet if condition just saying that if it new json then completed.